app/policies/policy_context.rb

```diff
@@ -0,0 +1,12 @@
+class PolicyContext
+  attr_reader :record
+
+  def initialize(record, user)
+    @record = record
+    @user = user
+  end
+
+  def policy_class
+    "#{@user.roles.uniq.first}DefenceRequestPolicy".classify
```

Comment:
Wouldn't it be better to have more explicit way of constructing these? That you'd have to request which policy do you want and for which user. Ie:

```ruby
def initialize(record, policy_type, user)
```

I think we have already identified that we'll have multiple roles for a user.